### PR TITLE
[AlloyV2] Support direct photo uploads as well as photo URLs

### PR DESCRIPTION
**Pull request by chrismytton**
Add support for upload_files in AlloyV2. This came up after putting Hackney Alloy highways live.

See https://github.com/mysociety/fixmystreet/pull/3590 for the related FixMyStreet fix.

### CODE REVIEW

**Review by dracos — APPROVED** [review]
One comment is all

File: perllib/Open311/Endpoint/Integration/AlloyV2.pm [hunk 1]
```diff
@@ -434,10 +434,10 @@ sub post_service_request_update {
         value => $updates
     }];
 
-    if ( $self->config->{resource_attachment_attribute_id} && @{ $args->{media_url} }) {
+    if ( $self->config->{resource_attachment_attribute_id} && ( @{ $args->{media_url} } // @{ $args->{uploads} } ) ) {
```
[the remark this answers is not in the document]
Comment (chrismytton) [reply perllib/Open311/Endpoint/Integration/AlloyV2.pm:437]: After discussing in Slack we realised this was masking a wider issue, which is that we don't currently support photo uploads for updates, either in open311-adapter or at the FixMyStreet end.

So need to drop the change on this line completely so it only uses `media_url` for updates.

[the remark this answers is not in the document]
Comment (chrismytton) [reply perllib/Open311/Endpoint/Integration/AlloyV2.pm]: Have force-pushed a fix, hope that's OK.

**Review by dracos — APPROVED** [review]

**Comment by chrismytton** [comment]
Tested on staging along with https://github.com/mysociety/fixmystreet/pull/3590 and it works 🎉 

Have deployed this change to productions now.

image

**Pull request merged**